FlowCrypt/Models/Common/Recipient.swift

```diff
@@ -22,4 +23,67 @@ extension Recipient {
         self.lastUsed = recipientObject.lastUsed
         self.pubKeys = recipientObject.pubKeys.map(PubKey.init)
     }
+
+    init(_ string: String) {
+        let parts = string.components(separatedBy: " ")
+
+        guard parts.count > 1, let email = parts.last else {
+            self.name = nil
+            self.email = string
+            return
+        }
+
+        self.email = email.filter { !["<", ">"].contains($0) }
+        let name = string
+            .replacingOccurrences(of: email, with: "")
+            .replacingOccurrences(of: "\"", with: "")
+            .trimmingCharacters(in: .whitespaces)
+        self.name = name == self.email ? nil : name
+    }
```

Comment:
This is way too dangerous to do by hand. You need to use an email parser that follows the RFC properly. You can't implement that in a few lines of code, there is a gazillion edge cases when parsing email strings.

Reply:
Which  library do you mean?
> Our MIME email library probably has a method for this, use that.